Marlin/src/gcode/control/M42.cpp

```diff
@@ -127,7 +127,7 @@ void GcodeSuite::M42() {
 
   #ifdef ARDUINO_ARCH_STM32
     // A simple I/O will be set to 0 by analogWrite()
-    if (pin_status <= 1) return;
+    if (pin_status <= 1 && !PWM_PIN(pin)) return;
```

Comment:
Is `&&` definitely wanted here, or would `||` be better?

Reply:
yes, its to avoid to put some pwm value on a normal pin, which turn it off... but allow 0 and 1 on pwm pins, 0 is useful :p
same as if (PWM_PIN(pin)) analogWrite(pin, value);